zkabacus-crypto/src/revlock.rs

```diff
@@ -222,20 +294,136 @@ mod test {
+
+        // Test that a specific invalid revocation secret produces the correct error
+        // invalid secret
+        let invalid_secret = UncheckedRevocationSecret { secret, index: 0 };
+        // invalid secret as bytes
+        let mut bytes = [0; 33];
+        bytes[0..32].copy_from_slice(&secret.to_bytes());
+
+        // generate lock corresponding to invalid secret using `from_raw` method
+        let invalid_digested = Sha3_256::digest(&bytes);
+        let invalid_lock = Scalar::from_raw([
+            u64::from_le_bytes(<[u8; 8]>::try_from(&invalid_digested[0..8]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&invalid_digested[8..16]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&invalid_digested[16..24]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&invalid_digested[24..32]).unwrap()),
+        ]);
+
+        assert!(matches!(
+            RevocationPair::try_from(UncheckedRevocationPair {
+                secret: invalid_secret,
+                lock: RevocationLock(invalid_lock)
+            }),
+            Err(Error::InvalidSecret())
+        ));
+
+        // Test that a specific valid revocation secret produces a verified pair
+        // valid secret
+        let valid_secret = UncheckedRevocationSecret { secret, index: 1 };
+        // valid secret as bytes
+        bytes = [1; 33];
+        bytes[0..32].copy_from_slice(&secret.to_bytes());
+        // generate lock corresponding to valid secret using `from_raw` method
+        let digested = Sha3_256::digest(&bytes);
+        let lock = Scalar::from_raw([
+            u64::from_le_bytes(<[u8; 8]>::try_from(&digested[0..8]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&digested[8..16]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&digested[16..24]).unwrap()),
+            u64::from_le_bytes(<[u8; 8]>::try_from(&digested[24..32]).unwrap()),
+        ]);
+
+        assert!(RevocationPair::try_from(UncheckedRevocationPair {
+            secret: valid_secret,
+            lock: RevocationLock(lock)
+        })
+        .is_ok());
+
+        // Test that a specific valid revocation secret with an invalid lock produces correct error
+        // test that giving an invalid lock with a valid secret errors
+        assert!(matches!(
+            RevocationPair::try_from(UncheckedRevocationPair {
+                secret: valid_secret,
+                lock: RevocationLock(invalid_lock)
+            }),
+            Err(Error::MismatchedPair())
+        ));
+    }
+
+    #[test]
+    pub fn run_serialize_deserialize() {
+        // Normal case
+        let mut rng = thread_rng();
+        for _ in 1..1000 {
+            let rp = RevocationPair::new(&mut rng);
+            let serialized_rp = bincode::serialize(&rp).unwrap();
+            let deserialized_rp = bincode::deserialize(&serialized_rp).unwrap();
```

Comment:
This causes `cargo test` to error because `bincode` isn't available unless you enable the `sqlite` feature flag. To fix:

```rust
#[test]
#[cfg(feature = "sqlite")]
pub fn ...
```